File: sdk/elixir/test/ltp/canonical_envelope_v1_test.exs
defmodule LTP.CanonicalEnvelopeV1Test do
  use ExUnit.Case, async: true

  @root Path.expand("../../../..", __DIR__)

  test "matches the shared golden vector" do
    fixture =
      @root
      |> Path.join("tests/security/canonical-envelope-v1.json")
      |> File.read!()
      |> Jason.decode!()

    expected =
      @root
      |> Path.join("tests/security/canonical-envelope-v1.txt")
      |> File.read!()
      |> String.trim_trailing()

    assert LTP.Crypto.serialize_canonical(fixture) == expected
  end

  test "rejects unsafe integers" do
    assert_raise ArgumentError, ~r/unsafe integer/, fn ->
      LTP.Crypto.serialize_canonical(%{"payload" => %{"unsafe" => 9_007_199_254_740_992}})
    end
  end
end

File: sdk/js/src/__tests__/canonicalEnvelopeV1.test.ts
import { readFileSync } from 'fs';
import { resolve } from 'path';
import { serializeCanonical } from '../crypto';

const root = resolve(__dirname, '../../../../');
const fixture = JSON.parse(readFileSync(resolve(root, 'tests/security/canonical-envelope-v1.json'), 'utf8'));
const expected = readFileSync(resolve(root, 'tests/security/canonical-envelope-v1.txt'), 'utf8').trimEnd();

const actual = serializeCanonical(fixture);
if (actual !== expected) {
  throw new Error(`Canonical Envelope v1 mismatch\nexpected: ${expected}\nactual:   ${actual}`);
}

let rejected = false;
try {
  serializeCanonical({ ...fixture, payload: { unsafe: Number.MAX_SAFE_INTEGER + 1 } });
} catch {
  rejected = true;
}
if (!rejected) {
  throw new Error('Canonical Envelope v1 must reject unsafe integers');
}

console.log('✓ Canonical Envelope v1 JavaScript golden vector');
